lib/dsprot/Makefile

```diff
@@ -0,0 +1,215 @@
+# ===================================================================
+# See doc/BUILD_OVERVIEW.TXT for a clearer outline of this build process
+# ===================================================================
+
+MAKEFLAGS += --no-print-directory
+
+MWCCVER := 2.0/sp2p3
+
+# Relevant directories
+BUILD_DIR  :=  ./build
+C_OBJ_DIR  :=  $(BUILD_DIR)/src
+SRC_DIR    :=  ./src
+INC_DIR    :=  ./include
+TOOL_DIR   :=  ./tools
+
+include ../../common.mk
+
+ifeq ($(NODEP),)
+  DEP_PARAM := -gccdep -MD
+  include $(wildcard $(C_OBJ_DIR)/*.d)
+else
+  DEP_PARAM := 
+endif
+
+INSTALL_PREFIX ?= ./install
+INSTALL_LIBDIR := $(INSTALL_PREFIX)/lib
+
+ELFCODER_DIR := $(TOOL_DIR)/elfcoder
+
+# Tools
+ELFCODER := $(ELFCODER_DIR)/build/elfcoder$(EXE)
+
+# C / ASM compilation parameters
+CC_PARAM   :=  -O4,p -enum int -proc arm946E -gccext,on -fp soft -lang c99 -char signed -inline on,noauto -Cpp_exceptions off -interworking -i $(INC_DIR) -c
+ASM_PARAM  :=  -proc arm5TE -i $(INC_DIR)
+LIB_PARAM  :=  -nostdlib -library
+
+CC_PARAM   +=  -W all -W pedantic -W noimpl_signedunsigned -W noimplicitconv -W nounusedarg -W nomissingreturn -W error
+
+# Output library file
+LIBRARY_NAME  :=  dsprot.a
+LIBRARY       :=  $(BUILD_DIR)/$(LIBRARY_NAME)
+
+# Files (in this specific order) that will go into the library
+LIBRARY_FILES := \
```

Comment:
imo this is *really* not a good idea
files should be hardcoded in the LSF for game code
nowhere else
the makefile should build all files found in the source directory and then the lsf should decide what to link and where
I really dislike this